packages/lit-dev-content/site/docs/frameworks/react.md

```diff
@@ -0,0 +1,205 @@
+
+The wrappers are targeted at two different audiences:
+- Users of web components can wrap components and controllers for their own use in their own React projects.
+- Vendors of components can publish React wrappers so that their React users have idiomatic versions of their components.
+
+### Why are wrappers needed?
+
+React can already render web components, since custom elements are just HTML elements and React knows how to render HTML. But React makes some assumptions about HTML elements that don't always hold for custom elements, and it treats lower-case tag names differently from upper-case component names in ways that can make custom elements harder than necessary to use.
+
+For instance, React assumes that all JSX properties map to HTML element attributes, and provides no way to set properties. This makes it difficult to pass complex data (like objects, arrays, or functions) to web components. React also assumes that all DOM events have corresponding "event properties" (`onclick`, `onmousemove`, etc), and uses those instead of calling `addEventListener()`. This means that to properly use more complex web components you often have to use `ref()` and imperative code. (For more information on the limitations of React's web component integration, see [Custom Elements Everywhere](https://custom-elements-everywhere.com/libraries/react/results/results.html).)
+
+React is working on fixes to these issues, but in the meantime, our wrappers take care of setting properties and listening to events for you.
+
+The `@lit-labs/react` package provides two main exports:
+
+-   `createComponent()` creates a React component that _wraps_ an existing web component. The wrapper allows you to set props on the component and add event listeners to the component like you would any other React component.
+
+-   `useController()` lets you use a Lit reactive controller as a React hook.  
+
+## createComponent
+
+The `createComponent()` function makes a React component wrapper for a custom element class. The wrapper correctly passes React `props` to properties accepted by the custom element and listens for events dispatched by the custom element.
+
+### Usage
+
+Import `React`, a custom element class, and `createComponent`.
+
+```js
+import React from 'react';
+import {createComponent} from '@lit-labs/react';
+import {MyElement} from './my-element.js';
+
+export const MyElementComponent = createComponent({
+  tagName: 'my-element',
+  elementClass: MyElement,
+  react: React,
+  events: {
+    onActivate: 'activate',
+    onChange: 'change',
+  },
+});
+```
+
+After defining the React component, you can use it just as you would any other React component.
+
+```jsx
+<MyElementComponent
+  active={isActive}
+  onActivate={(e) => setIsActive(e.active)}
+  onChange={handleChange}
+/>
+```
+
+{% aside "positive" "no-header" %}
```

Comment:
I like this aside!

I wonder, can we have a standard "Try it" phrasing and some kind of different icon for playground links? @aomarks @e111077 if you have opinions there.

Reply:
Why not just do:

`{% playground-ide "examples/react-basics/" %}` and embed it directly

Reply:
I had not considered that possibility! Is it fine to reference the same example in the doc while leaving it standalone in the playgrounds section too? I don't know if we have any precedence for that.

Reply:
/shrug it should also stay up to date. Though there's a thing that goes through and essentially server renders the playground previews in the docs section. Not sure if that is limited to the `packages/lit-dev-content/samples/docs` folder

Reply:
I'll keep them as links to playground in an aside for now. Embedding the existing example is adding too much noise in my eyes when there are already code blocks right above it. I think it would be good future improvement to replace the code blocks with a tailored example.